codegen/src/lintdoc.rs

```diff
@@ -1608,16 +1647,24 @@ fn write_action<L: ServiceLanguage>(
 /// Parse and analyze the provided code block, and asserts that it emits
 /// exactly zero or one diagnostic depending on the value of `expect_diagnostic`.
 /// That diagnostic is then emitted as text into the `content` buffer
+#[allow(clippy::too_many_arguments)]
 fn print_diagnostics_or_actions(
     group: &'static str,
     rule: &'static str,
     test: &CodeBlockTest,
     code: &str,
     config: &Option<Configuration>,
     buffer: &mut HTML<&mut Vec<u8>>,
+    file_system: &HashMap<String, String>,
     to_print_kind: ToPrintKind,
 ) -> Result<()> {
-    let file_path = format!("code-block.{}", test.tag);
+    let file_path = {
+        if let Some(file_path) = &test.file_path {
+            normalize_file_path(file_path)
+        } else {
+            format!("code-block.{}", test.tag)
```

Comment:
We don't validate if the file has or doesn't have an extension. Do we check for that in the main repository ?